src/automata/RequestResponseRequester.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#include "RequestResponseRequester.h"
+
+#include <iostream>
+
+namespace reactivesocket {
+
+void RequestResponseRequesterBase::subscribe(Subscriber<Payload>& subscriber) {
+  DCHECK(!consumingSubscriber_);
+  consumingSubscriber_.reset(&subscriber);
+  // FIXME
+  // Subscriber::onSubscribe is delivered externally, as it may attempt to
+  // synchronously deliver Subscriber::request.
+  // Q: why do we want to avoid Subscriber::request being delivered
+  // synchronously?
+}
+
+void RequestResponseRequesterBase::onNext(Payload request) {
+  switch (state_) {
+    case State::NEW: {
+      state_ = State::REQUESTED;
+      Frame_REQUEST_RESPONSE frame(
+          streamId_, FrameFlags_EMPTY, std::move(std::move(request)));
+      connection_->outputFrameOrEnqueue(frame.serializeOut());
+      break;
+    }
+    case State::REQUESTED:
+      break;
```

Comment:
I think for State::REQUESTED we can error out, however State::CANCEL seems possible if the subscriber cancels the subscription immediately. e.g. 

 ReactiveSocket::requestResponse() 
     ->Subscriber::onSubscribe() calls (Subscription)RequestResponseRequester::cancel()
    -> RequestResponseRequester::onNext() // state is canceled